hub/src/server/authentication.js

```diff
@@ -306,3 +372,49 @@ export function validateAuthorizationHeader(authHeader: string, serverName: stri
   const challengeText = getChallengeText(serverName)
   return authObject.isAuthenticationValid(address, challengeText, { validHubUrls, requireCorrectHubUrl })
 }
+
+
+/*
+ * Get the authentication scopes from the authorization header.
+ * Does not check the authorization header or its association token
+ * (do that with validateAuthorizationHeader first).
+ *
+ * Returns the scopes on success
+ * Throws on malformed auth header
+ */
+export function getAuthenticationScopes(authHeader: string) {
+  const authObject = parseAuthHeader(authHeader)
+  return authObject.getAuthenticationScopes()
+}
+
+
+/*
+ * Validate authentication scopes.  They must be well-formed,
+ * and there can't be too many of them.
+ * Return true if valid.
+ * Throw ValidationError on error
+ */
+function validateScopes(scopes: Array<AuthScopeType>) {
+  if (scopes.length > 8) {
+    throw new ValidationError('Too many authentication scopes')
+  }
+
+  for (let i = 0; i < scopes.length; i++) {
+    const scope = scopes[i]
+    
+    // valid scope?
+    let valid = false
+    for (let j = 0; j < AuthScopes.length; j++) {
```

Comment:
This code could be a bit simpler if you use `.find()` instead of this inner loop